prime-angular-frontend/src/app/modules/site-registration/shared/components/summary-card/summary-card.component.ts

```diff
@@ -1,12 +1,36 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, Output, EventEmitter, Input } from '@angular/core';
+import { Template } from '@angular/compiler/src/render3/r3_ast';
 
 @Component({
   selector: 'app-summary-card',
   templateUrl: './summary-card.component.html',
   styleUrls: ['./summary-card.component.scss']
 })
 export class SummaryCardComponent implements OnInit {
-  constructor() { }
+  @Input() public icon: string;
+  @Input() public title: string;
+  @Input() public menu: Template;
+  @Input() public menuOutletContext: object;
```

Comment:
We might want to make this typing `object | null` it's quite possible that you could pass in a `menu` template reference, but not supply a `menuOutletContext` since the template is scoped to the parent.  We could then set a default in the constructor of `null`.

Reply:
The only other issue I find with `object` is that it means any non-primitive types so by saying `object` instead of `{ [key:string]: any }` which is equivalent you're sort of saying I'm okay with an object literal or an array.  While `{ [key:string]: any }` allows the same, but you're sort of saying I'm more likely expecting an object literal.